openmc/tallies.py

```diff
@@ -2210,12 +2249,9 @@ def __add__(self, other):
             new_tally.with_summary = self.with_summary
             new_tally.num_realization = self.num_realizations
 
-            for self_filter in self.filters:
-                new_tally.add_filter(self_filter)
-            for nuclide in self.nuclides:
-                new_tally.add_nuclide(nuclide)
-            for score in self.scores:
-                new_tally.add_score(score)
+            new_tally.filters = self.filters
```

Comment:
These lists need to be deep copied here. Otherwise if the user (or tally arithmetic) changes the scores in `self` or `new_tally` it will erroneously change them in the other.